File: app/src/main/java/com/mahendri/pasbeli/ui/MainModule.java
package com.mahendri.pasbeli.ui;

import android.widget.ProgressBar;

import dagger.Module;
import dagger.Provides;

/**
 * @author Mahendri
 */

@Module
public class MainModule {

    @Provides
    ProgressBar provideProgressBar (MainActivity mainActivity) {
        return new ProgressBar(mainActivity, null, android.R.attr.progressBarStyle);
    }

}

File: app/src/main/java/com/mahendri/pasbeli/ui/main/MainModule.kt
package com.mahendri.pasbeli.ui.main

import android.widget.ProgressBar

import dagger.Module
import dagger.Provides

/**
 * @author Mahendri
 */

@Module
class MainModule {

    @Provides
    internal fun provideProgressBar(mainActivity: MainActivity): ProgressBar {
        return ProgressBar(mainActivity, null, android.R.attr.progressBarStyle)
    }

}
